fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java

```diff
@@ -2722,14 +2723,13 @@ public void changeCatalogDb(ConnectContext ctx, String identifier) throws DdlExc
         String[] parts = identifier.split("\\.");
         if (parts.length != 1 && parts.length != 2) {
             ErrorReport.reportDdlException(ErrorCode.ERR_BAD_CATALOG_ERROR, identifier);
-        } else if (parts.length == 1) {
-            dbName = catalogMgr.isInternalCatalog(currentCatalogName) ?
-                    ClusterNamespace.getFullName(ctx.getClusterName(), identifier) : identifier;
         } else {
-            String newCatalogName = parts[0];
+            String newCatalogName = parts.length == 1 ?
```

Comment:
If current_catalog is external_catalog, we use another db under external db. it will convert to internal catalog.